File: apps/disclosure-rag/test_db_connection.ts
/**
 * TypeScript test script for Ultraterrestrial Database Connector
 * Demonstrates database operations and analytics
 */

import { createUltraterrestrialDB, demoUltraterrestrialDB, generateEmbedding } from './lib/connectors/ultraterrestrial_db';

async function testConnection() {
    console.log('🔌 Testing TypeScript database connection...');
    
    const db = createUltraterrestrialDB();
    
    try {
        const connected = await db.connect();
        
        if (connected) {
            console.log('✅ Database connection successful!');
            
            // Get database statistics
            console.log('\n📊 Database statistics:');
            const stats = await db.getDatabaseStats();
            Object.entries(stats).forEach(([key, value]) => {
                console.log(`  • ${key}: ${value}`);
            });
            
            // Test keyword search
            console.log('\n🔍 Testing keyword search...');
            const searchResults = await db.keywordSearch('navy pilots UFO encounter', 'events', 3);
            
            if (searchResults.length > 0) {
                console.log('Search results:');
                searchResults.forEach(result => {
                    console.log(`  • ${result.title} (score: ${result.similarityScore.toFixed(3)})`);
                });
            } else {
                console.log('No search results found');
            }
            
            // Test analytics
            console.log('\n📈 Testing analytics...');
            const influence = await db.getPersonnelInfluence(3);
            if (influence.length > 0) {
                console.log('Top influential personnel:');
                influence.forEach(person => {
                    console.log(`  • ${person.name}: ${person.influenceScore.toFixed(1)}`);
                });
            }
            
            const hotspots = await db.getUAPHotspots(3);
            if (hotspots.length > 0) {
                console.log('\nTop UAP hotspots:');
                hotspots.forEach(spot => {
                    console.log(`  • ${spot.location}: ${spot.sightingCount} sightings`);
                });
            }
            
        } else {
            console.log('❌ Database connection failed!');
        }
        
    } catch (error) {
        console.error('❌ Test failed:', error);
        console.log('\nMake sure you have:');
        console.log('1. PostgreSQL running with pgvector extension');
        console.log('2. Database "ultraterrestrial" created');
        console.log('3. Schema loaded from schema_ultraterrestrial_pgvector.sql');
        console.log('4. Required npm packages: npm install pg @types/pg');
    } finally {
        await db.close();
    }
}

async function testEmbeddingGeneration() {
    console.log('\n🔤 Testing embedding generation...');
    
    const apiKey = process.env.OPENAI_API_KEY;
    if (!apiKey) {
        console.log('⚠️  OPENAI_API_KEY not found in environment');
        return;
    }
    
    try {
        const embedding = await generateEmbedding(
            'USS Nimitz UFO encounter with Tic Tac object',
            apiKey
        );
        
        if (embedding) {
            console.log(`✅ Generated embedding with ${embedding.length} dimensions`);
            console.log(`First 5 values: [${embedding.slice(0, 5).map(v => v.toFixed(4)).join(', ')}...]`);
        } else {
            console.log('❌ Failed to generate embedding');
        }
    } catch (error) {
        console.error('❌ Embedding generation failed:', error);
    }
}

async function runDemo() {
    console.log('\n🎬 Running full TypeScript demo...');
    
    try {
        await demoUltraterrestrialDB();
    } catch (error) {
        console.error('❌ Demo failed:', error);
    }
}

async function main() {
    console.log('🚀 Ultraterrestrial TypeScript Database Test');
    console.log('=' .repeat(50));
    
    // Test basic connection and operations
    await testConnection();
    
    // Test embedding generation if API key available
    await testEmbeddingGeneration();
    
    // Run full demo
    await runDemo();
    
    console.log('\n✅ TypeScript tests completed!');
}

// Run if this is the main module
if (require.main === module) {
    main().catch(console.error);
}

export { testConnection, testEmbeddingGeneration, runDemo };

File: apps/disclosure-rag/test_db_connection.py
#!/usr/bin/env python3
"""
Test script for Ultraterrestrial PostgreSQL + pgvector Database
Demonstrates CSV import and analytics capabilities
"""

import asyncio
import sys
import os
from pathlib import Path

# Add the lib directory to Python path
sys.path.append(str(Path(__file__).parent / 'lib' / 'connectors'))

try:
    from ultraterrestrial_db import UltraterrestrialDB, quick_setup_and_import, demo_search_and_analytics
except ImportError as e:
    print(f"❌ Import error: {e}")
    print("Make sure you have the required dependencies installed:")
    print("pip install asyncpg pandas numpy sentence-transformers")
    sys.exit(1)

async def test_connection_only():
    """Test database connection without importing data"""
    print("🔌 Testing database connection...")
    
    db = UltraterrestrialDB()
    
    try:
        await db.connect()
        print("✅ Database connection successful!")
        
        # Get basic stats
        stats = await db.get_database_stats()
        print("\n📊 Current database state:")
        for key, value in stats.items():
            print(f"  • {key}: {value}")
        
    except Exception as e:
        print(f"❌ Connection failed: {e}")
        print("\nMake sure you have:")
        print("1. PostgreSQL running with pgvector extension")
        print("2. Database 'ultraterrestrial' created")
        print("3. Schema loaded from schema_ultraterrestrial_pgvector.sql")
    finally:
        await db.close()

async def test_csv_import():
    """Test CSV import from exports directory"""
    print("📂 Testing CSV import from exports directory...")
    
    exports_dir = "./exports"
    if not Path(exports_dir).exists():
        print(f"❌ Exports directory not found: {exports_dir}")
        return
    
    try:
        db = await quick_setup_and_import(exports_dir)
        
        print("\n📊 Post-import database stats:")
        stats = await db.get_database_stats()
        for key, value in stats.items():
            print(f"  • {key}: {value}")
        
        # Test semantic search if embeddings are available
        print("\n🔍 Testing semantic search...")
        results = await db.semantic_search("navy pilots UFO encounter", "events", 3)
        
        if results:
            print("Search results:")
            for result in results:
                print(f"  • {result.title} (similarity: {result.similarity_score:.3f})")
        else:
            print("No results found (embeddings may not be generated yet)")
        
        await db.close()
        
    except Exception as e:
        print(f"❌ CSV import failed: {e}")

async def test_analytics():
    """Test analytics queries"""
    print("📈 Testing analytics capabilities...")
    
    try:
        await demo_search_and_analytics()
    except Exception as e:
        print(f"❌ Analytics test failed: {e}")

async def main():
    """Main test runner"""
    print("🚀 Ultraterrestrial Database Test Suite")
    print("=" * 50)
    
    # Check if CSV files exist
    exports_path = Path("./exports")
    csv_files = list(exports_path.glob("*.csv")) if exports_path.exists() else []
    
    print(f"\n📁 Found {len(csv_files)} CSV files in exports directory")
    if csv_files:
        print("Available files:")
        for csv_file in csv_files[:5]:  # Show first 5
            print(f"  • {csv_file.name}")
        if len(csv_files) > 5:
            print(f"  ... and {len(csv_files) - 5} more")
    
    # Test connection first
    await test_connection_only()
    
    # Ask user what to test
    print("\n" + "=" * 50)
    print("Choose test to run:")
    print("1. Connection test only (done above)")
    print("2. CSV import test")
    print("3. Analytics demo")
    print("4. All tests")
    
    try:
        choice = input("\nEnter choice (1-4): ").strip()
        
        if choice == "2":
            await test_csv_import()
        elif choice == "3":
            await test_analytics()
        elif choice == "4":
            await test_csv_import()
            await test_analytics()
        else:
            print("✅ Connection test completed!")
    
    except KeyboardInterrupt:
        print("\n👋 Test interrupted by user")
    except Exception as e:
        print(f"❌ Test failed: {e}")

if __name__ == "__main__":
    asyncio.run(main())